base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/commands/ConfigHandler.java

```diff
@@ -91,4 +96,51 @@ protected IWorkbenchWindow activeWindow() {
      */
 	public abstract void canWriteChanged(boolean canWrite);
 	
+    /**
+     * Implements the required eclipse interface.
+     * 
+     * Declared as final here, implement executeAndCatchErrors instead so that
+     * we can "hook into" any errors that occur in any of these dialogs and do
+     * something sensible with them.
+     * 
+     * @param event
+     *            the execution event
+     * @return null (ignored by eclipse framework)
+     */
+    @Override
+    public final Object execute(ExecutionEvent event) {
+        try {
+            safeExecute(event);
+        } catch (RuntimeException e) {
+            onError(e);
+        }
+        return null;
+    }
+
+    /**
+     * Executes an eclipse event. Any runtime errors thrown by this method will
+     * be caught by the execute() method and cause the onError() handler to be
+     * called.
+     * 
+     * @param event
+     *            the eclipse event
+     */
+    public abstract void safeExecute(ExecutionEvent event);
+
+    /**
+     * Error handler. Override this to provide specific error handling
+     * behaviour.
+     * 
+     * @param ex
+     *            the exception that was thrown from the handler
+     */
+    public void onError(Exception ex) {
+        Logger.getGlobal().log(Level.SEVERE, "Exception while executing configserver command", ex);
```

Comment:
This doesn't seem to work. I get:

```
SEVERE: Exception while executing configserver command
java.lang.NullPointerException
	at uk.ac.stfc.isis.ibex.ui.configserver.commands.helpers.EditConfigHelper.openDialog(EditConfigHelper.java:57)
	at uk.ac.stfc.isis.ibex.ui.configserver.commands.helpers.ConfigHelper.createDialog(ConfigHelper.java:33)
	at uk.ac.stfc.isis.ibex.ui.configserver.commands.EditConfigHandler.safeExecute(EditConfigHandler.java:59)
	at uk.ac.stfc.isis.ibex.ui.configserver.commands.ConfigHandler.execute(ConfigHandler.java:113)
	at org.eclipse.ui.internal.handlers.HandlerProxy.execute(HandlerProxy.java:290)
	at org.eclipse.core.commands.Command.executeWithChecks(Command.java:499)
	at org.eclipse.core.commands.ParameterizedCommand.executeWithChecks(ParameterizedCommand.java:508)
	at org.eclipse.ui.internal.handlers.HandlerService.executeCommand(HandlerService.java:169)
	at org.eclipse.ui.internal.handlers.SlaveHandlerService.executeCommand(SlaveHandlerService.java:241)
	at org.eclipse.ui.menus.CommandContributionItem.handleWidgetSelection(CommandContributionItem.java:829)
	at org.eclipse.ui.menus.CommandContributionItem.access$19(CommandContributionItem.java:815)
	at org.eclipse.ui.menus.CommandContributionItem$5.handleEvent(CommandContributionItem.java:805)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:84)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1053)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4169)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3758)
	at org.eclipse.ui.internal.Workbench.runEventLoop(Workbench.java:2701)
	at org.eclipse.ui.internal.Workbench.runUI(Workbench.java:2665)
	at org.eclipse.ui.internal.Workbench.access$4(Workbench.java:2499)
	at org.eclipse.ui.internal.Workbench$7.run(Workbench.java:679)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:332)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:668)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:149)
	at uk.ac.stfc.isis.ibex.product.Application.start(Application.java:40)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:110)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:79)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:353)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:180)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:629)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:584)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1438)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1414)
```

which doesn't give the original site of the exception and there is nothing in About->consoleLog. I am not quite sure what it should be. To reproduce add the line `ioc = null;` at line 199 in EditableConfiguration.